File: OneCodeTeam/How-to-log-UI-traces-into-Windows-Azure-table-29443/[VB]-How to log UI traces into Windows Azure table storage/C#/CSAzureJSLogging/AzureClient.cs
/****************************** Module Header ******************************\
* Module Name: HomeController.cs
* Project:     CSAzureJSLogging
* Copyright (c) Microsoft Corporation
*
* HomeController
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\*****************************************************************************/
using Microsoft.WindowsAzure.Storage;
using Microsoft.WindowsAzure.Storage.Shared.Protocol;
using Microsoft.WindowsAzure.Storage.Table;
using System.Collections.Generic;

namespace CSAzureJSLogging
{
    public class AzureClient
    {
       
        public static CloudStorageAccount StorageAccount = new CloudStorageAccount(new Microsoft.WindowsAzure.Storage.Auth.StorageCredentials("{Your-Storage-Account}", "{Your-Storage-Acccount-Key}"), false);

        /// <summary>
        /// Enable Table Service CORS
        /// Make sure log table can be use.
        /// </summary>
        public static void TableServiceInitialization() 
        {
            CloudTableClient tableClient = StorageAccount.CreateCloudTableClient();

            // Aslo check the log table make sure it has been created.
            CloudTable table = tableClient.GetTableReference("log");
            table.CreateIfNotExists();
            ServiceProperties tableServiceProperties = new ServiceProperties();
            tableServiceProperties.HourMetrics = null;
            tableServiceProperties.MinuteMetrics = null;
            tableServiceProperties.Logging = null;

            // Enable and Configure CORS
            ConfigureCors(tableServiceProperties);
            tableClient.SetServiceProperties(tableServiceProperties);
        }

        /// <summary>
        /// Adds CORS rule to the service properties.
        /// </summary>
        /// <param name="serviceProperties">ServiceProperties</param>
        private static void ConfigureCors(ServiceProperties serviceProperties)
        {
            serviceProperties.Cors = new CorsProperties();
            serviceProperties.Cors.CorsRules.Add(new CorsRule()
            {
                AllowedHeaders = new List<string>() { "*" },
                AllowedMethods = CorsHttpMethods.Put | CorsHttpMethods.Get | CorsHttpMethods.Head | CorsHttpMethods.Post,
                AllowedOrigins = new List<string>() { "*" },
                ExposedHeaders = new List<string>() { "*" },
                MaxAgeInSeconds = 1800 // 30 minutes
            });
        }
    }
}

File: OneCodeTeam/How-to-log-UI-traces-into-Windows-Azure-table-29443/[VB]-How to log UI traces into Windows Azure table storage/VB/VBAzureJSLogging/AzureClient.vb
'***************************** Module Header ******************************\
' Module Name: HomeController.vb
' Project:     VBAzureJSLogging
' Copyright (c) Microsoft Corporation
'
' AzureClient
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'****************************************************************************/
Imports Microsoft.WindowsAzure.Storage.Table
Imports Microsoft.WindowsAzure.Storage
Imports Microsoft.WindowsAzure.Storage.Shared.Protocol

Public Class AzureClient

    Public Shared StorageAccount As New CloudStorageAccount(New Microsoft.WindowsAzure.Storage.Auth.StorageCredentials("{Your-Storage-Account}", "{Your-Storage-Acccount-Key}"), False)

    ''' <summary>
    ''' Enable Table Service CORS
    ''' Make sure log table can be use.
    ''' </summary>
    Public Shared Sub TableServiceInitialization()
        Dim tableClient As CloudTableClient = StorageAccount.CreateCloudTableClient()
        ' Aslo check the log table make sure it has been created.
        Dim table As CloudTable = tableClient.GetTableReference("log")
        table.CreateIfNotExists()
        Dim tableServiceProperties As New ServiceProperties()
        tableServiceProperties.HourMetrics = Nothing
        tableServiceProperties.MinuteMetrics = Nothing
        tableServiceProperties.Logging = Nothing

        ' Enable and Configure CORS
        ConfigureCors(tableServiceProperties)
        tableClient.SetServiceProperties(tableServiceProperties)
    End Sub

    ''' <summary>
    ''' Adds CORS rule to the service properties.
    ''' </summary>
    ''' <param name="serviceProperties">ServiceProperties</param>
    Private Shared Sub ConfigureCors(serviceProperties As ServiceProperties)
        serviceProperties.Cors = New CorsProperties()
        ' 30 minutes
        serviceProperties.Cors.CorsRules.Add(New CorsRule() With { _
             .AllowedHeaders = New List(Of String)() From { _
                "*" _
            }, _
             .AllowedMethods = CorsHttpMethods.Put Or CorsHttpMethods.[Get] Or CorsHttpMethods.Head Or CorsHttpMethods.Post, _
             .AllowedOrigins = New List(Of String)() From { _
                "*" _
            }, _
             .ExposedHeaders = New List(Of String)() From { _
                "*" _
            }, _
             .MaxAgeInSeconds = 1800 _
        })
    End Sub
End Class
